test/testdata/rbi/hash.rb

```diff
@@ -87,3 +87,10 @@
   T.assert_type!(v, Float)
   T.assert_type!(i, Integer)
 end
+
+T.assert_type!({}.any?, T::Boolean)
+T.assert_type!({ a: 'ant', b: 'bear', c: 'cat' }.any?(/d/), T::Boolean)
+T.assert_type!(
+  { a: 'ant', b: 'bear', c: 'cat' }.any? { |key, value| key == value },
+  T::Boolean
+)
```

Comment:
```suggestion
T.reveal_type({ a: 'ant', b: 'bear', c: 'cat' }.any? do |key, value| # error: Revealed type: `T::Boolean`
  T.reveal_type(key) # error: Revealed type: `Symbol`
  T.reveal_type(key) # error: Revealed type: `String`
end)
T.reveal_type({ a: 'ant', b: 'bear', c: 'cat' }.any? do |(key, value)| # error: Revealed type: `T::Boolean`
  T.reveal_type(key) # error: Revealed type: `Symbol`
  T.reveal_type(key) # error: Revealed type: `String`
end)
```

Reply:
The second part of this test (using `|(key, value)|` fails until the above suggestion on the other file is applied.

Reply:
The `reveal_type` assertions fail for me here as it's not able to correctly type key or value.

Example:

```
test/testdata/rbi/hash.rb:94: ERROR: Expected error of form:
94:   T.reveal_type(key) # error: Revealed type: `Symbol`
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Revealed type: `Symbol`
Found error:
94:   T.reveal_type(key) # error: Revealed type: `Symbol`
      ^^^^^^^^^^^^^^^^^^ error: Revealed type: `T.untyped`
```

Reply:
The new commit posted solves this issue for me. Thanks for that!